src/khoj/database/adapters/__init__.py

```diff
@@ -446,12 +446,27 @@ async def set_user_github_config(user: KhojUser, pat_token: str, repos: list):
     return config
 
 
-def get_user_search_model_or_default(user=None):
-    if user and UserSearchModelConfig.objects.filter(user=user).exists():
-        return UserSearchModelConfig.objects.filter(user=user).first().setting
+def get_default_search_model() -> SearchModelConfig:
+    default_search_model = SearchModelConfig.objects.filter(name="default").first()
 
-    if SearchModelConfig.objects.filter(name="default").exists():
-        return SearchModelConfig.objects.filter(name="default").first()
+    if default_search_model:
+        return default_search_model
+    else:
+        SearchModelConfig.objects.create()
+
+    return SearchModelConfig.objects.first()
+
+
+def get_user_default_search_model(user: KhojUser = None) -> SearchModelConfig:
+    if user:
+        user_search_model = UserSearchModelConfig.objects.filter(user=user).first()
+        if user_search_model:
+            return user_search_model.setting
+
+    default_search_model = SearchModelConfig.objects.filter(name="default").first()
+
+    if default_search_model:
+        return default_search_model
     else:
         SearchModelConfig.objects.create()
```

Comment:
nit: This can be replaced with get_default_search_model to dedupe logic
```suggestion
    return get_default_search_model()
```